packages/ifpack2/src/Ifpack2_BlockRelaxation_def.hpp

```diff
@@ -547,45 +553,43 @@ initialize ()
     IsParallel_ = false;
   }
 
-  ++NumInitialize_;
-  Time_->stop ();
-  InitializeTime_ += Time_->totalElapsedTime ();
-  IsInitialized_ = true;
-}
-
-
-template<class MatrixType,class ContainerType>
-void
-BlockRelaxation<MatrixType,ContainerType>::
-computeBlockCrs ()
-{
-  // typedef Tpetra::Map<local_ordinal_type,global_ordinal_type,node_type>     map_type; // unused
-  // typedef Tpetra::Import<local_ordinal_type,global_ordinal_type, node_type> import_type; // unused
-  using Teuchos::Ptr;
-  using Teuchos::RCP;
-  using Teuchos::rcp;
-  using Teuchos::Array;
-  using Teuchos::ArrayView;
-
-  Time_->start (true);
-
-  // reset values
-  IsComputed_ = false;
+  // We should have checked for this in setParameters(), so it's a
+  // logic_error, not an invalid_argument or runtime_error.
+  TEUCHOS_TEST_FOR_EXCEPTION
+    (NumSweeps_ < 0, std::logic_error, "Ifpack2::BlockRelaxation::initialize: "
+     "NumSweeps_ = " << NumSweeps_ << " < 0.");
 
   // Extract the submatrices
-  ExtractSubmatrices ();
+  ExtractSubmatricesStructure ();
 
   // Compute the weight vector if we're doing overlapped Jacobi (and
   // only if we're doing overlapped Jacobi).
-  TEUCHOS_TEST_FOR_EXCEPTION
-    (PrecType_ == Ifpack2::Details::JACOBI && OverlapLevel_ > 0, std::runtime_error,
-     "Ifpack2::BlockRelaxation::computeBlockCrs: "
-     "We do not support overlapped Jacobi yet for Tpetra::BlockCrsMatrix.  Sorry!");
+  if (PrecType_ == Ifpack2::Details::JACOBI && OverlapLevel_ > 0) {
+    TEUCHOS_TEST_FOR_EXCEPTION
+      (hasBlockCrsMatrix_, std::runtime_error,
+       "Ifpack2::BlockRelaxation::initialize: "
+       "We do not support overlapped Jacobi yet for Tpetra::BlockCrsMatrix.  Sorry!");      
 
-  ++NumCompute_;
+    // weight of each vertex
+    W_ = rcp (new vector_type (A_->getRowMap ()));
+    W_->putScalar (STS::zero ());
+    Teuchos::ArrayRCP<scalar_type > w_ptr = W_->getDataNonConst(0);
```

Comment:
@mhoemmen, I don't object, but this code is touched in the commit only because of indenting (brace nesting) change. I'd like to keep changes for issues orthogonal to the line smoother out of this PR.